api/v1beta2/ratelimitpolicy_types.go

```diff
@@ -261,6 +274,20 @@ func (r *RateLimitPolicy) DirectReferenceAnnotationName() string {
 	return RateLimitPolicyDirectReferenceAnnotationName
 }
 
+func (r *RateLimitPolicy) CommonSpec() *RateLimitPolicyCommonSpec {
+	if r.Spec.Defaults != nil {
+		return r.Spec.Defaults
+	}
+
+	return &r.Spec.RateLimitPolicyCommonSpec
+}
+
+// GetLimits returns the limits based on whether default or implicit rules are defined.
+// Default rules takes precedence
+func (r *RateLimitPolicy) GetLimits() map[string]Limit {
+	return r.CommonSpec().Limits
+}
```

Comment:
Both functions are not required. It's to keep inline with what was done with AuthPolicy defaults:
https://github.com/Kuadrant/kuadrant-operator/blob/b4d066d8aa83f772fb14116a1c861fdef25942bb/api/v1beta2/authpolicy_types.go#L327-L351

I feel it mainly reads nicer in any calling functions that the `CommonSpec` is abstracted away by using the `GetX` functions rather than using `r.GetCommonSpec().X` in code outside the `/api` package with some benefit with abstracting out changes to the spec in general. 

Personally I'd prefer to keep this but can update if you feel strongly about this

Reply:
I had not seen that example yet but now that I have. Should either `GetCommonSpec` function exist on the struct. You could write the `GetCommonSpec` as a single function that takes in a generic of the RLP or AP and return the correct type. 

I do feel strongly about this but not yet ready to die on that hill. Over my "break" I had time to think about stuff. Some of that came from the book "A philosophy of software design", good book worth a read. Two points from the book scrams at me when I seen the code we are talking about are; Modules should be deep not wide, and design it twice. Then I have also being looking into cyclomatic complexity and the affects that can have on a project and its developers. Main take is modules should be deep not wide.

If you are going to keep the code the way it is, can you document what the `CommonSpec` does so that LSP's can give better context.

Reply:
Updated following some offline discussion. Removed unncessary getters and added coment for `CommonSpec`